packages/@ember/routing/route.ts

```diff
@@ -1233,14 +1243,24 @@ class Route<Model = unknown> extends EmberObject.extend(ActionHandler, Evented)
     @param {Object} value the value passed to find
     @private
   */
-  findModel(...args: any[]) {
-    // SAFETY: it's all absurd lies; there is no explicit contract for `store`
-    // and we allow people to register *anything* here and we call it: GLHF! The
-    // fallback path here means we correctly handle the case where there is no
-    // explicit store injection on the route subclass.
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let store = ('store' in this ? this.store : get(this, '_store')) as any;
-    return store.find(...args);
+  findModel(type: string, value: unknown) {
+    deprecate(
+      `The implicit model loading behavior for routes is deprecated. ` +
+        `Please define an explicit model hook for ${this.fullRouteName}.`,
+      false,
+      {
+        id: 'deprecate-implicit-route-model',
```

Comment:
I see there is a deprecation guide -- I think that should be here in the `url` property?

Reply:
@kategengler => https://github.com/emberjs/ember.js/pull/20524.